module/entities/helper.js

```diff
@@ -1,4 +1,943 @@
+ * @param {params} search - search entity data. 
+ * @param {params} limit - page limit. 
+ * @param {params} pageNo - page no. 
+ * @returns {Array} - List of all sub list entities.
+ */
+
+    static subEntityList(entities, entityId, type, search, limit, pageNo) {
+        return new Promise(async (resolve, reject) => {
+
+            try {
+
+                let result = [];
+                let obj = {
+                    entityId: entityId,
+                    type: type,
+                    search: search,
+                    limit: limit,
+                    pageNo: pageNo
+                }
+
+                if (entityId !== "") {
+                    result = await this.subEntities(
+                        obj
+                    );
+                } else {
+
+                    await Promise.all(entities.map(async (entity) => {
+
+                        obj["entityId"] = entity;
+                        let entitiesDocument = await this.subEntities(
+                            obj
+                        );
+
+                        if (Array.isArray(entitiesDocument.data) &&
+                            entitiesDocument.data.length > 0
+                        ) {
+                            result = entitiesDocument;
+                        }
+                    }));
+                }
+
+                if (result.data && result.data.length > 0) {
+                    result.data = result.data.map(data => {
+
+                        let cloneData = { ...data };
+                        // cloneData["label"] = cloneData.name;
+                        // cloneData["_id"] = cloneData._id;
+                        cloneData['address'] = cloneData.addressLine1;
+                        if (cloneData.addressLine1) {
+                            delete cloneData.addressLine1;
+                        }
+                        return cloneData;
+                    })
+                }
+
+                let columns = _subEntityListColumns();
+                resolve({
+                    message: constants.apiResponses.ENTITIES_FETCHED,
+                    result: {
+                        count: result.count ? result.count : 0,
+                        columns: columns,
+                        data: result.data
+                    }
+                });
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+    /**
+   * Get either immediate entities or entity traversal based upon the type.
+   * @method
+   * @name subEntities
+   * @param {body} entitiesData
+   * @returns {Array} - List of all immediate entities or traversal data.
+   */
+
+    static subEntities(entitiesData) {
+        return new Promise(async (resolve, reject) => {
+
+            try {
+
+                let entitiesDocument;
+
+                if (entitiesData.type !== "") {
+
+                    entitiesDocument = await this.entityTraversal(
```

Comment:
@rakeshSgr - We can merge the 2 into 1 function entityTraversal and immediateEntities, if in immediateEntities the type is not given, pick the first element of childHierarchyPath.